### PR TITLE
[System test] Deposit liquidation using multiple partial take offers

### PR DESCRIPTION
**Pull request by tomaszslabon**
This pull request:
- adds a system test
- removes the folder with integration tests - we can add all high level tests to the folder with system tests.

Refs: https://github.com/keep-network/coverage-pools/issues/83

### CODE REVIEW

**Comment by dimpar** [comment]
I think it is better to change these tests to align with other system tests, ie. using `init-contract.js` etc. They will check the full cycle when partially taking offers.

**Comment by tomaszslabon** [comment]
> I think it is better to change these tests to align with other system tests, ie. using `init-contract.js` etc. They will check the full cycle when partially taking offers.

Ok, I will change it to use real contracts.

File: test/system/liquidation-multiple-take-offers.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      it("should not exist", async () => {
+        const auctionAddress = await riskManagerV1.depositToAuction(
+          tbtcDeposit1.address
+        )
+        expect(auctionAddress).to.be.equal(ZERO_ADDRESS)
+      })
+    })
+  })
+
+  describe("when auction has been fully filled will multiple partial fills", () => {
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply test/system/liquidation-multiple-take-offers.test.js]: Done in dc5d6e631fb050aea4c306d13754dc87f73e9b08.

File: test/system/liquidation-multiple-take-offers.test.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      )
+
+      // Deposit has been liquidated.
+      expect(await tbtcDeposit1.currentState()).to.equal(11) // LIQUIDATED
+
+      // Signer bonds should land on the risk manager contract.
+      await expect(tx).to.changeEtherBalance(riskManagerV1, bondedAmount)
+    })
+
+    it("should transfer collateral tokens to the first bidder", async () => {
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply test/system/liquidation-multiple-take-offers.test.js]: Done in dc5d6e631fb050aea4c306d13754dc87f73e9b08.

File: test/system/liquidation-multiple-take-offers.test.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    let bidder2InitCollateralBalance
+
+    before(async () => {
+      await tbtcDeposit1.notifyRedemptionSignatureTimedOut()
+      await riskManagerV1.notifyLiquidation(tbtcDeposit1.address)
+
+      const auctionAddress = await riskManagerV1.depositToAuction(
+        tbtcDeposit1.address
+      )
+      auction = new ethers.Contract(auctionAddress, Auction.abi, bidder1)
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply test/system/liquidation-multiple-take-offers.test.js]: Good idea, done in 8afc9accf1d642ca584096a806a649f50bba68c3.

File: test/system/liquidation-multiple-take-offers.test.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      await auction.connect(bidder1).takeOffer(lotSize.mul(40).div(100))
+      tx = await auction.connect(bidder2).takeOffer(lotSize.mul(60).div(100))
+    })
+
+    it("should close auction", async () => {
+      expect(await auction.isOpen()).to.be.false
+    })
+
+    it("should liquidate the deposit", async () => {
+      // Bidders spent their TBTC.
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply test/system/liquidation-multiple-take-offers.test.js]: I split this one big it into smaller ones, as they seem to test three different things:
```
    it("should decrease the amount of TBTC for bidders", async () => {
      const bidder1CurrentBalance = await tbtcToken.balanceOf(bidder1.address)
      const bidder2CurrentBalance = await tbtcToken.balanceOf(bidder2.address)
      expect(bidder1InitialBalance.sub(bidder1CurrentBalance)).to.be.equal(
        lotSize.mul(40).div(100)
      )
      expect(bidder2InitialBalance.sub(bidder2CurrentBalance)).to.be.equal(
        lotSize.mul(60).div(100)
      )
    })

    it("should liquidate the deposit", async () => {
      expect(await tbtcDeposit1.currentState()).to.equal(11) // LIQUIDATED
    })

    it("should transfer ether from signer bonds to risk manager", async () => {
      await expect(tx).to.changeEtherBalance(riskManagerV1, bondedAmount)
    })
```
Done in 729b75547e4fb0d52171e6303d0f1ceee01dc023.

File: test/system/liquidation-multiple-take-offers.test.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+    before(async () => {
+      await tbtcDeposit1.notifyRedemptionSignatureTimedOut()
+      await riskManagerV1.notifyLiquidation(tbtcDeposit1.address)
+
+      const auctionAddress = await riskManagerV1.depositToAuction(
+        tbtcDeposit1.address
+      )
+      auction = new ethers.Contract(auctionAddress, Auction.abi, bidder1)
+      await tbtcToken.connect(bidder1).approve(auction.address, lotSize)
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply test/system/liquidation-multiple-take-offers.test.js]: Sure, done in 88505624fb93fcbf047d2bda670de5ae13bf85df.

**Review by dimpar — APPROVED** [review]

**Pull request merged**